### PR TITLE
feat: distributed execute gRPC and Prometheus query in Frontend

### PR DESCRIPTION
**Pull request by MichaelScofield**
Make Frontend supports gRPC and Prometheus read/write, execute in distribute.

### CODE REVIEW

File: config/datanode.example.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 node_id = 42
 mode = 'distributed'
-rpc_addr = '0.0.0.0:3001'
+rpc_addr = '127.0.0.01:3001'
```
Comment (v0y4g3r) [inline config/datanode.example.toml:3]:
```suggestion
rpc_addr = '127.0.0.1:3001'
```

File: config/datanode.example.toml [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 wal_dir = '/tmp/greptimedb/wal'
 rpc_runtime_size = 8
-mysql_addr = '0.0.0.0:3306'
+mysql_addr = '127.0.0.01:3306'
```
Comment (v0y4g3r) [inline config/datanode.example.toml:6]:
```suggestion
mysql_addr = '127.0.0.1:3306'
```

File: config/frontend.example.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 mode = 'distributed'
 datanode_rpc_addr = '127.0.0.1:3001'
-http_addr = '0.0.0.0:4000'
+http_addr = '127.0.0.01:4000'
```
Comment (v0y4g3r) [inline config/frontend.example.toml:3]:
```suggestion
http_addr = '127.0.0.1:4000'
```

File: config/metasrv.example.toml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-bind_addr = '0.0.0.0:3002'
-server_addr = '0.0.0.0:3002'
+bind_addr = '127.0.0.01:3002'
+server_addr = '127.0.0.01:3002'
```
Comment (v0y4g3r) [inline config/metasrv.example.toml:2]:
```suggestion
bind_addr = '127.0.0.1:3002'
server_addr = '127.0.0.1:3002'
```

File: config/standalone.example.toml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 node_id = 0
 mode = 'standalone'
-http_addr = '0.0.0.0:4000'
-datanode_mysql_addr = '0.0.0.0:3306'
+http_addr = '127.0.0.01:4000'
+datanode_mysql_addr = '127.0.0.01:3306'
```
Comment (v0y4g3r) [inline config/standalone.example.toml:4]:
```suggestion
http_addr = '127.0.0.1:4000'
datanode_mysql_addr = '127.0.0.1:3306'
```

File: config/standalone.example.toml [hunk 6]
```diff
@@ -10,25 +10,25 @@ type = 'File'
 data_dir = '/tmp/greptimedb/data/'
 
 [grpc_options]
-addr = '0.0.0.0:4001'
+addr = '127.0.0.01:4001'
```
Comment (v0y4g3r) [inline config/standalone.example.toml:13]:
```suggestion
addr = '127.0.0.1:4001'
```

File: config/standalone.example.toml [hunk 7]
```diff
@@ -10,25 +10,25 @@ type = 'File'
...
 runtime_size = 8
 
 [mysql_options]
-addr = '0.0.0.0:4002'
+addr = '127.0.0.01:4002'
```
Comment (v0y4g3r) [inline config/standalone.example.toml:17]:
```suggestion
addr = '127.0.0.1:4002'
```

File: config/standalone.example.toml [hunk 8]
```diff
@@ -10,25 +10,25 @@ type = 'File'
...
 runtime_size = 2
 
 [influxdb_options]
 enable = true
 
 [opentsdb_options]
-addr = '0.0.0.0:4242'
+addr = '127.0.0.01:4242'
```
Comment (v0y4g3r) [inline config/standalone.example.toml:24]:
```suggestion
addr = '127.0.0.1:4242'
```

File: config/standalone.example.toml [hunk 9]
```diff
@@ -10,25 +10,25 @@ type = 'File'
...
 enable = true
 runtime_size = 2
 
 [prometheus_options]
 enable = true
 
 [postgres_options]
-addr = '0.0.0.0:4003'
+addr = '127.0.0.01:4003'
```
Comment (v0y4g3r) [inline config/standalone.example.toml:32]:
```suggestion
addr = '127.0.0.1:4003'
```

File: src/cmd/src/datanode.rs [hunk 10]
```diff
@@ -145,9 +145,9 @@ mod tests {
             )),
         };
         let options: DatanodeOptions = cmd.try_into().unwrap();
-        assert_eq!("0.0.0.0:3001".to_string(), options.rpc_addr);
+        assert_eq!("127.0.0.01:3001".to_string(), options.rpc_addr);
```
Comment (v0y4g3r) [inline src/cmd/src/datanode.rs:148]:
```suggestion
        assert_eq!("127.0.0.1:3001".to_string(), options.rpc_addr);
```

File: src/cmd/src/datanode.rs [hunk 11]
```diff
@@ -145,9 +145,9 @@ mod tests {
...
         assert_eq!("/tmp/greptimedb/wal".to_string(), options.wal_dir);
-        assert_eq!("0.0.0.0:3306".to_string(), options.mysql_addr);
+        assert_eq!("127.0.0.01:3306".to_string(), options.mysql_addr);
```
Comment (v0y4g3r) [inline src/cmd/src/datanode.rs:150]:
```suggestion
        assert_eq!("127.0.0.1:3306".to_string(), options.mysql_addr);
```

File: src/cmd/src/metasrv.rs [hunk 12]
```diff
@@ -104,13 +104,13 @@ mod tests {
     fn test_read_from_cmd() {
         let cmd = StartCommand {
             bind_addr: Some("127.0.0.1:3002".to_string()),
-            server_addr: Some("0.0.0.0:3002".to_string()),
+            server_addr: Some("127.0.0.01:3002".to_string()),
```
Comment (v0y4g3r) [inline src/cmd/src/metasrv.rs:107]:
```suggestion
            server_addr: Some("127.0.0.1:3002".to_string()),
```

File: src/cmd/src/metasrv.rs [hunk 13]
```diff
@@ -104,13 +104,13 @@ mod tests {
...
             store_addr: Some("127.0.0.1:2380".to_string()),
             config_file: None,
         };
         let options: MetaSrvOptions = cmd.try_into().unwrap();
         assert_eq!("127.0.0.1:3002".to_string(), options.bind_addr);
-        assert_eq!("0.0.0.0:3002".to_string(), options.server_addr);
+        assert_eq!("127.0.0.01:3002".to_string(), options.server_addr);
```
Comment (v0y4g3r) [inline src/cmd/src/metasrv.rs:113]:
```suggestion
        assert_eq!("127.0.0.1:3002".to_string(), options.server_addr);
```

File: src/cmd/src/metasrv.rs [hunk 14]
```diff
@@ -126,8 +126,8 @@ mod tests {
             )),
         };
         let options: MetaSrvOptions = cmd.try_into().unwrap();
-        assert_eq!("0.0.0.0:3002".to_string(), options.bind_addr);
-        assert_eq!("0.0.0.0:3002".to_string(), options.server_addr);
+        assert_eq!("127.0.0.01:3002".to_string(), options.bind_addr);
+        assert_eq!("127.0.0.01:3002".to_string(), options.server_addr);
```
Comment (v0y4g3r) [inline src/cmd/src/metasrv.rs:130]:
```suggestion
        assert_eq!("127.0.0.1:3002".to_string(), options.bind_addr);
        assert_eq!("127.0.0.1:3002".to_string(), options.server_addr);
```

**Review by v0y4g3r — COMMENTED** [review]
Please also change all the `127.0.0.01` to ``127.0.0.1``

File: config/datanode.example.toml (hunk 1 above)
Comment (killme2008) [reply config/datanode.example.toml:3]: It's not fixed

File: config/datanode.example.toml (hunk 2 above)
Comment (killme2008) [reply config/datanode.example.toml:6]: It's not fixed too.

File: config/frontend.example.toml (hunk 3 above)
Comment (killme2008) [reply config/frontend.example.toml:3]: It's not fixed

**Review by killme2008 — COMMENTED** [review]
Please fix the wrong addresses in TOML files.

**Pull request merged**